x-pack/plugins/reporting_notifier/public/plugin.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as Rx from 'rxjs';
+import { catchError, filter, map, mergeMap } from 'rxjs/operators';
+import { i18n } from '@kbn/i18n';
+import {
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  PluginInitializerContext,
+} from '../../../../src/core/public';
+import {
+  JOB_COMPLETION_NOTIFICATIONS_POLLER_CONFIG,
+  JOB_COMPLETION_NOTIFICATIONS_SESSION_KEY,
+} from '../constants';
+import { JobId, JobStatusBuckets } from '../index.d';
+import { getGeneralErrorToast } from './components';
+import { ReportingNotifierStreamHandler as StreamHandler } from './lib/stream_handler';
+
+const {
+  jobCompletionNotifier: { interval: JOBS_REFRESH_INTERVAL },
+} = JOB_COMPLETION_NOTIFICATIONS_POLLER_CONFIG;
+
+function getStored(): JobId[] {
+  const sessionValue = sessionStorage.getItem(JOB_COMPLETION_NOTIFICATIONS_SESSION_KEY);
+  return sessionValue ? JSON.parse(sessionValue) : [];
+}
+
+export class ReportingNotifierPublicPlugin implements Plugin<any, any> {
+  private poller$: Rx.Observable<JobStatusBuckets> | null = null;
+
+  // FIXME: License checking: only active, non-expired licenses allowed
+  constructor(initializerContext: PluginInitializerContext) {}
+
+  public setup(core: CoreSetup) {}
+
+  // FIXME: only perform these actions for authenticated routes
```

Comment:
@elastic/kibana-platform any pointers on where I could look to take care of the fixmes ?

Reply:
This FIXME seems to depend on NP changes to provide an "unauthenticated paths" service. Some details in https://github.com/elastic/kibana/pull/39477

Reply:
How does this behave now? Will it throw errors when unauthenticated?

Reply:
It's unlikely to do anything when unauthenticated, because the HTTP calls are only made if pending job IDs exist in sessionStorage, which *seems* to be emptied on logout, and would not be populated in a fresh new session.

I don't think that totally makes it moot though. If I recall, the security plugin handles clearing sessionStorage when the user logs out, so we are getting by on assuming that happens. If it isn't, then error toast messages could pop up on the screen.